### PR TITLE
Fixing proxied images for federated posts.

### LINKED ISSUE #4736: [Bug]: Federated thumbnails from unproxied -> proxied server not working
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

I'm testing out image proxying + federation between DS9 and voyager, which have full domain names.

server | setting
--- | ---
DS9 | StoreLinkPreviews
Voyager | ProxyAllImages

- DS9 appears to generate thumbnails (locally) correctly for a federated post: https://ds9.lemmy.ml/post/2191
- Voyager's local posts thumbnail proxying works fine: https://voyager.lemmy.ml/post/2012065
- Voyager's federated posts thumbnail proxying is not: https://voyager.lemmy.ml/post/2012064
  - It appears to wrongly be using the post.url for the thumbnail link, rather than the metadata image.

### Steps to Reproduce

1. Create a link post with a thumbnail from ds9 (with store all link previews turned on)
2. Go to the post on voyager, the thumbnail is wrong.

### Technical Details

NA

### Version

0.19.4-rc.1

### Lemmy Instance URL

all

### PR DESCRIPTION
**Pull request by dessalines**
- Also added test.
- Fixes #4736

### CODE REVIEW

File: Cargo.toml
Comment (dessalines) [inline Cargo.toml:70]: New clippy warnings for these on nightly.

File: api_tests/src/image.spec.ts
Comment (dessalines) [inline api_tests/src/image.spec.ts:265]: This was the key test that failed before the fix below was added.

File: crates/apub/src/objects/post.rs
Comment (dessalines) [inline crates/apub/src/objects/post.rs:231]: Culprit was here, these urls are no longer proxied, but also it was proxying even for non-image posts.

**Review by SleeplessOne1917 — APPROVED** [review]
Nothing looks off to me, but I'm not too familiar with the image proxying or federation parts of the code, so I will defer to @Nutomic @phiresky @dullbananas .

File: Cargo.toml
Comment (Nutomic) [reply Cargo.toml:70]: What does the priority  = -1 do?

File: api_tests/src/image.spec.ts
Comment (Nutomic) [inline api_tests/src/image.spec.ts:251]: Duplicate comment

File: api_tests/src/image.spec.ts
Comment (Nutomic) [inline api_tests/src/image.spec.ts:226]: > Thumbnail of remote image link is proxied if setting enabled

**Review by dullbananas — APPROVED** [review]

File: Cargo.toml
Comment (dessalines) [reply Cargo.toml:70]: It overrides the group / collection lints. https://rust-lang.github.io/rust-clippy/master/index.html#/lint_groups_priority

**Comment by dessalines** [comment]
After this is merged, I'll make a new RC, and do more testing on the voyager and DS9.

**Pull request merged**
